src/player.cpp

```diff
@@ -157,6 +162,12 @@ void Player::clearHud()
 	}
 }
 
+u16 Player::getMaxHotbarItemcount()
+{
+	InventoryList *mainlist = inventory.getList("main");
+	return mainlist ? std::min((s32) mainlist->getSize(), hud_hotbar_itemcount) : 0;
```

Comment:
make more sense to cast the list size to u16 here IMO

Reply:
Done in d5039d509